### PR TITLE
26162 merge common code

### LINKED ISSUE #26158: Telemetry-1.0 Disabled warning message
The warning message "The MicroProfile Telemetry Tracing feature is enabled but not configured to generate traces for the {0} application." does not appear when MicroProfile Telemetry 1.0 is not enabled.

**Steps to Reproduce**  
1. Enable the mpTelemetry-1.0 feature
2. Do not set "otel.sdk.disabled=false"
3. Make request to application 
The warning message shows up as `CWMOT5100.tracing.is.disabled` 

**Expected behavior**  
"The MicroProfile Telemetry Tracing feature is enabled but not configured to generate traces for the {0} application." should be in the logs"

**Diagnostic information:**  
 - OpenLiberty Version: 23.0.0.9
 - Affected feature(s): mpTelemetry-1.0
 - server.xml configuration (WITHOUT sensitive information like passwords):
 ```	
 <featureManager>
		<feature>mpTelemetry-1.0</feature>
</featureManager>
```

### LINKED ISSUE #26162: Remove code duplication in open telemetry
We have three projects 

- io.openliberty.microprofile.telemetry.1.0.internal
- io.openliberty.microprofile.telemetry.1.1.internal
- io.openliberty.microprofile.telemetry.common

Common code in the first two should be moved to the third.

### PR DESCRIPTION
**Pull request by benjamin-confino**
fixes #26162
fixes #26158

### CODE REVIEW

**Comment by benjamin-confino** [comment]
#help

**Comment by benjamin-confino** [comment]
#build
#spawn.fullfat.buckets=io.openliberty.microprofile.telemetry.1.0.internal_fat,io.openliberty.microprofile.telemetry.1.0.internal_fat_tck,io.openliberty.microprofile.telemetry.1.0.internal.jaeger_fat,io.openliberty.microprofile.telemetry.1.1.internal_fat_tck

**Comment by benjamin-confino** [comment]
#build
#spawn.fullfat.buckets=io.openliberty.microprofile.telemetry.1.0.internal_fat,io.openliberty.microprofile.telemetry.1.0.internal_fat_tck,io.openliberty.microprofile.telemetry.1.0.internal.jaeger_fat,io.openliberty.microprofile.telemetry.1.1.internal_fat_tck

**Comment by benjamin-confino** [comment]
#build

**Comment by benjamin-confino** [comment]
#build

File: dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/publish/servers/Telemetry10JaxWithConcurrency/server.xml [hunk 1]
```diff
@@ -45,6 +45,6 @@
         <classloader apiTypeVisibility="+third-party"/>
     </application>
 
-    <!--logging traceSpecification="TELEMETRY=all:RESTfulWS=all"/-->
+    <logging traceSpecification="*=info:JCDI=all:com.ibm.ws.webbeans*=all:org.apache.webbeans*=all:org.jboss.weld*=all:com.ibm.ws.cdi*=all"/>
```
Comment (benjamin-confino) [inline dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/publish/servers/Telemetry10JaxWithConcurrency/server.xml]: Undo

**Comment by benjamin-confino** [comment]
#build

File: dev/com.ibm.ws.cdi.extension_fat/fat/src/com/ibm/ws/cdi/extension/apps/spi/SPIExtensionServlet.java [hunk 2]
```diff
@@ -15,8 +15,13 @@
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.fail;
 
+import com.ibm.ws.cdi.extension.tests.CDIExtensionRepeatActions;
```
Comment (tevans78) [inline dev/com.ibm.ws.cdi.extension_fat/fat/src/com/ibm/ws/cdi/extension/apps/spi/SPIExtensionServlet.java:18]: Might be a good idea to split this new CDI test into a separate commit or even a separate PR?

File: dev/io.openliberty.microprofile.telemetry.1.0.internal/bnd.bnd [hunk 3]
```diff
@@ -51,17 +53,20 @@ Service-Component: \
 # Need an explicit import for api.common because it gets used by weld proxies
 # created for other API classes
 Import-Package: \
-	io.opentelemetry.api.common,\
-	io.openliberty.microprofile.telemetry.common.*,\
-	*
+  com.ibm.websphere.ras,\
+  com.ibm.websphere.ras.annotation,\
+  io.opentelemetry.api.common,\
+  io.openliberty.microprofile.telemetry.common.*,\
```
Comment (tevans78) [inline dev/io.openliberty.microprofile.telemetry.1.0.internal/bnd.bnd]: I think this package should have been `io.openliberty.microprofile.telemetry.internal.common`

File: dev/io.openliberty.microprofile.telemetry.1.0.internal/src/io/openliberty/microprofile/telemetry/internal/cdi/OpenTelemetryProducer.java [hunk 4]
```diff
@@ -1,226 +1,66 @@
...
- * are made available under the terms of the Eclipse Public License 2.0
- * which accompanies this distribution, and is available at
- * http://www.eclipse.org/legal/epl-2.0/
- *
- * SPDX-License-Identifier: EPL-2.0
- *
- * Contributors:
- *     IBM Corporation - initial API and implementation
- *******************************************************************************/
 package io.openliberty.microprofile.telemetry.internal.cdi;
```
Comment (tevans78) [inline dev/io.openliberty.microprofile.telemetry.1.0.internal/src/io/openliberty/microprofile/telemetry/internal/cdi/OpenTelemetryProducer.java]: Where did the copyright header go?!

File: dev/io.openliberty.microprofile.telemetry.1.0.internal/src/io/openliberty/microprofile/telemetry/internal/cdi/OpenTelemetryProducer.java [hunk 5]
```diff
@@ -1,226 +1,66 @@
...
+        return super.getTracer(openTelemetry);
     }
 
+    @Override
     @Produces
     @ApplicationScoped
     public Span getSpan() {
-        return new SpanProxy();
+        return super.getSpan();
     }
```
Comment (tevans78) [inline dev/io.openliberty.microprofile.telemetry.1.0.internal/src/io/openliberty/microprofile/telemetry/internal/cdi/OpenTelemetryProducer.java]: Why is this an override which then calls super?

File: dev/io.openliberty.microprofile.telemetry.1.0.internal/src/io/openliberty/microprofile/telemetry/internal/cdi/SPIMetaData.java [hunk 6]
```diff
@@ -23,7 +23,7 @@
 import com.ibm.ws.cdi.extension.CDIExtensionMetadataInternal;
 
 import io.openliberty.cdi.spi.CDIExtensionMetadata;
-import io.openliberty.microprofile.telemetry.internal.helper.AgentDetection;
+import io.openliberty.microprofile.telemetry.common.internal.helper.AgentDetection;
```
Comment (tevans78) [inline dev/io.openliberty.microprofile.telemetry.1.0.internal/src/io/openliberty/microprofile/telemetry/internal/cdi/SPIMetaData.java]: I would have named this package `io.openliberty.microprofile.telemetry.internal.common` ... no need for the `helper` subpackage.

File: dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/fat/src/io/openliberty/microprofile/telemetry/internal_fat/FATSuite.java [hunk 7]
```diff
@@ -18,26 +18,7 @@
 @RunWith(Suite.class)
 @MinimumJavaLevel(javaLevel = 11)
 @SuiteClasses({
-                JaxRsIntegration.class,
                 JaxRsIntegrationWithConcurrency.class,
```
Comment (tevans78) [inline dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/fat/src/io/openliberty/microprofile/telemetry/internal_fat/FATSuite.java]: Where did all the other tests go?

File: dev/io.openliberty.microprofile.telemetry.1.1.internal/bnd.bnd [hunk 8]
```diff
@@ -54,11 +54,13 @@ Import-Package: \
   *
 
 Export-Package: \
-  io.openliberty.microprofile.telemetry.internal.rest;version=1.1;thread-context=true
-
+	io.openliberty.microprofile.telemetry.internal.cdi;version='1.0',\
+	io.openliberty.microprofile.telemetry.resources;version='1.0',\
+	io.openliberty.microprofile.telemetry.rest;version='1.0',\
```
Comment (tevans78) [inline dev/io.openliberty.microprofile.telemetry.1.1.internal/bnd.bnd]: I can't immediately find anything in a package called `io.openliberty.microprofile.telemetry.rest`. I can see the internal version but not this (which I would interpret as an API package)

File: dev/io.openliberty.microprofile.telemetry.1.0.internal/src/io/openliberty/microprofile/telemetry/internal/cdi/OpenTelemetryProducer.java (hunk 5 above)
Comment (benjamin-confino) [reply dev/io.openliberty.microprofile.telemetry.1.0.internal/src/io/openliberty/microprofile/telemetry/internal/cdi/OpenTelemetryProducer.java]: CDI producers are not inherited (if they were you'd get an ambigious resolution between the super and subclass). So those method exist just to be annotated.

File: dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/fat/src/io/openliberty/microprofile/telemetry/internal_fat/FATSuite.java (hunk 7 above)
Comment (benjamin-confino) [reply dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/fat/src/io/openliberty/microprofile/telemetry/internal_fat/FATSuite.java]: Oops. That change is Left over from debugging.

File: dev/com.ibm.ws.cdi.extension_fat/fat/src/com/ibm/ws/cdi/extension/apps/spi/SPIExtensionServlet.java (hunk 2 above)
Comment (benjamin-confino) [reply dev/com.ibm.ws.cdi.extension_fat/fat/src/com/ibm/ws/cdi/extension/apps/spi/SPIExtensionServlet.java:18]: It is in a separate commit

File: dev/io.openliberty.microprofile.telemetry.1.1.internal/src/io/openliberty/microprofile/telemetry/internal/cdi/OpenTelemetryProducer.java [hunk 9]
```diff
@@ -1,204 +1,68 @@
...
- * are made available under the terms of the Eclipse Public License 2.0
- * which accompanies this distribution, and is available at
- * http://www.eclipse.org/legal/epl-2.0/
- *
- * SPDX-License-Identifier: EPL-2.0
- *
- * Contributors:
- *     IBM Corporation - initial API and implementation
- *******************************************************************************/
 package io.openliberty.microprofile.telemetry.internal.cdi;
```
Comment (tevans78) [inline dev/io.openliberty.microprofile.telemetry.1.1.internal/src/io/openliberty/microprofile/telemetry/internal/cdi/OpenTelemetryProducer.java]: Copyright?

File: dev/io.openliberty.io.opentelemetry.1.29/bnd.bnd [hunk 10]
```diff
@@ -15,32 +15,32 @@ bVersion=1.0
...
+	zipkin2.internal,\
+	zipkin2.reporter,\
+	zipkin2.reporter.okhttp3,\
+	jakarta.interceptor,\
+	jakarta.enterprise.util,\
+	com.ibm.wsspi.classloading,\
+	org.osgi.framework,\
+	com.ibm.websphere.ras,\
+	com.ibm.websphere.ras.annotation,\
+	io.openliberty.microprofile.telemetry.internal.common.resources
```
Comment (tevans78) [inline dev/io.openliberty.io.opentelemetry.1.29/bnd.bnd]: We removed the bundle dependency on common so we should probably remove the import too

File: dev/io.openliberty.io.opentelemetry/bnd.bnd [hunk 11]
```diff
@@ -16,28 +16,28 @@ src: src, resources
...
+	zipkin2.internal,\
+	zipkin2.reporter,\
+	zipkin2.reporter.okhttp3,\
+	jakarta.interceptor,\
+	jakarta.enterprise.util,\
+	com.ibm.wsspi.classloading,\
+	org.osgi.framework,\
+	com.ibm.websphere.ras,\
+	com.ibm.websphere.ras.annotation,\
+	io.openliberty.microprofile.telemetry.internal.common.resources
```
Comment (tevans78) [inline dev/io.openliberty.io.opentelemetry/bnd.bnd:40]: remove this import

File: dev/io.openliberty.microprofile.telemetry.1.1.internal/src/io/openliberty/microprofile/telemetry11/internal/cdi/OpenTelemetryProducer.java [hunk 12]
```diff
@@ -0,0 +1,68 @@
+package io.openliberty.microprofile.telemetry11.internal.cdi;
```
Comment (tevans78) [inline dev/io.openliberty.microprofile.telemetry.1.1.internal/src/io/openliberty/microprofile/telemetry11/internal/cdi/OpenTelemetryProducer.java:11]: Copyright header

File: dev/com.ibm.ws.cdi.extension_fat/test-bundles/cdi.spi.extension/src/com/ibm/ws/cdi/extension/spi/test/bundle/getclass/beaninjection/MyStringSubClass.java [hunk 13]
```diff
@@ -0,0 +1,32 @@
+/*******************************************************************************
+ * Copyright (c) 2020, 2021 IBM Corporation and others.
```
Comment (tevans78) [inline dev/com.ibm.ws.cdi.extension_fat/test-bundles/cdi.spi.extension/src/com/ibm/ws/cdi/extension/spi/test/bundle/getclass/beaninjection/MyStringSubClass.java]: is this the correct date?

File: dev/io.openliberty.microprofile.telemetry.1.1.internal/src/io/openliberty/microprofile/telemetry11/internal/cdi/WithSpanInterceptor.java [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+ * Copyright (c) 2023 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
```
Comment (tevans78) [inline dev/io.openliberty.microprofile.telemetry.1.1.internal/src/io/openliberty/microprofile/telemetry11/internal/cdi/WithSpanInterceptor.java:11]: generally, you don't need to include the Contributors in the copyright header any more

File: dev/io.openliberty.microprofile.telemetry.internal.common/bnd.bnd [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+# SPDX-License-Identifier: EPL-2.0
+#*******************************************************************************
+-include= ~../cnf/resources/bnd/bundle.props
+bVersion=1.0
+
+Bundle-Name: io.openliberty.microprofile.telemetry.internal.common
+Bundle-SymbolicName: io.openliberty.microprofile.telemetry.internal.common
+Bundle-Description: MicroProfile.telemetry, common code
+
+Require-Capability: osgi.ee; filter:="(&(osgi.ee=JavaSE)(version=11))"
```
Comment (tevans78) [inline dev/io.openliberty.microprofile.telemetry.internal.common/bnd.bnd:17]: I thought we added the `javac.source` and `javac.target` back in?

File: dev/io.openliberty.microprofile.telemetry.1.0.internal/src/io/openliberty/microprofile/telemetry10/internal/cdi/OpenTelemetryProducer.java [hunk 16]
```diff
@@ -0,0 +1,78 @@
...
+import io.openliberty.microprofile.telemetry.internal.common.cdi.OpenTelemetryInfo;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.baggage.Baggage;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdk;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.inject.Produces;
+
+public class OpenTelemetryProducer extends AbstractOpenTelemetryProducer {
```
Comment (tevans78) [inline dev/io.openliberty.microprofile.telemetry.1.0.internal/src/io/openliberty/microprofile/telemetry10/internal/cdi/OpenTelemetryProducer.java:29]: A bit more javadoc would be useful

File: dev/io.openliberty.microprofile.telemetry.1.0.internal/src/io/openliberty/microprofile/telemetry10/internal/cdi/WithSpanInterceptor.java [hunk 17]
```diff
@@ -0,0 +1,40 @@
...
+import io.opentelemetry.instrumentation.annotations.WithSpan;
+import io.opentelemetry.instrumentation.api.annotation.support.MethodSpanAttributesExtractor;
+import jakarta.annotation.Priority;
+import jakarta.inject.Inject;
+import jakarta.interceptor.Interceptor;
+
+@WithSpan
+@Interceptor
+@Priority(100)
+public class WithSpanInterceptor extends AbstractWithSpanInterceptor {
```
Comment (tevans78) [inline dev/io.openliberty.microprofile.telemetry.1.0.internal/src/io/openliberty/microprofile/telemetry10/internal/cdi/WithSpanInterceptor.java:25]: javadoc and/or comments?

File: dev/io.openliberty.microprofile.telemetry.1.1.internal/src/io/openliberty/microprofile/telemetry11/internal/cdi/OpenTelemetryProducer.java [hunk 18]
```diff
@@ -0,0 +1,68 @@
...
+import io.openliberty.microprofile.telemetry.internal.common.cdi.SpanProxy;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.baggage.Baggage;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdk;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.inject.Produces;
+
+public class OpenTelemetryProducer extends AbstractOpenTelemetryProducer {
```
Comment (tevans78) [inline dev/io.openliberty.microprofile.telemetry.1.1.internal/src/io/openliberty/microprofile/telemetry11/internal/cdi/OpenTelemetryProducer.java:29]: javadoc and/or comments?

File: dev/io.openliberty.microprofile.telemetry.1.1.internal/src/io/openliberty/microprofile/telemetry11/internal/cdi/WithSpanInterceptor.java [hunk 19]
```diff
@@ -0,0 +1,40 @@
...
+import io.opentelemetry.instrumentation.annotations.WithSpan;
+import io.opentelemetry.instrumentation.api.annotation.support.MethodSpanAttributesExtractor;
+import jakarta.annotation.Priority;
+import jakarta.inject.Inject;
+import jakarta.interceptor.Interceptor;
+
+@WithSpan
+@Interceptor
+@Priority(100)
+public class WithSpanInterceptor extends AbstractWithSpanInterceptor {
```
Comment (tevans78) [inline dev/io.openliberty.microprofile.telemetry.1.1.internal/src/io/openliberty/microprofile/telemetry11/internal/cdi/WithSpanInterceptor.java:27]: javadoc and/or comments?

File: dev/io.openliberty.microprofile.telemetry.internal.common/src/io/openliberty/microprofile/telemetry/internal/common/cdi/AbstractWithSpanInterceptor.java [hunk 20]
```diff
@@ -0,0 +1,151 @@
...
+import io.opentelemetry.instrumentation.api.instrumenter.InstrumenterBuilder;
+import io.opentelemetry.instrumentation.api.instrumenter.SpanNameExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.util.SpanNames;
+import jakarta.annotation.Priority;
+import jakarta.inject.Inject;
+import jakarta.interceptor.AroundInvoke;
+import jakarta.interceptor.Interceptor;
+import jakarta.interceptor.InvocationContext;
+
+public abstract class AbstractWithSpanInterceptor {
```
Comment (tevans78) [inline dev/io.openliberty.microprofile.telemetry.internal.common/src/io/openliberty/microprofile/telemetry/internal/common/cdi/AbstractWithSpanInterceptor.java:44]: javadoc and/or comments?

File: dev/io.openliberty.microprofile.telemetry.internal.common/src/io/openliberty/microprofile/telemetry/internal/common/rest/AbstractTelemetryContainerFilter.java [hunk 21]
```diff
@@ -0,0 +1,18 @@
...
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package io.openliberty.microprofile.telemetry.internal.common.rest;
+
+public abstract class AbstractTelemetryContainerFilter {
```
Comment (tevans78) [inline dev/io.openliberty.microprofile.telemetry.internal.common/src/io/openliberty/microprofile/telemetry/internal/common/rest/AbstractTelemetryContainerFilter.java:12]: What is the point in this abstract class? Just to hold one constant?

Comment (benjamin-confino) [reply dev/io.openliberty.microprofile.telemetry.internal.common/src/io/openliberty/microprofile/telemetry/internal/common/rest/AbstractTelemetryContainerFilter.java:12]: It is, TelemetryServletRequestListener references that constant so it has to be in common.

**Comment by benjamin-confino** [comment]
#build
#spawn.fullfat.buckets=io.openliberty.microprofile.telemetry.1.0.internal_fat,io.openliberty.microprofile.telemetry.1.0.internal_fat_tck,io.openliberty.microprofile.telemetry.1.0.internal.jaeger_fat,io.openliberty.microprofile.telemetry.1.1.internal_fat_tck

**Comment by Yas-Franco** [comment]
```
Task :io.openliberty.mpTelemetry.1.1.thirdparty:jar FAILED

error  : io.openliberty.io.opentelemetry.1.1;version=latest Not found
```

The buildpath reference to `io.openliberty.io.opentelemetry.1.1` needs to change to `io.openliberty.io.opentelemetry.1.29` [here](https://github.com/OpenLiberty/open-liberty/blob/integration/dev/io.openliberty.mpTelemetry.1.1.thirdparty/bnd.bnd#L48)

**Comment by benjamin-confino** [comment]
Thank you Yasmin

**Comment by benjamin-confino** [comment]
#build
#spawn.fullfat.buckets=io.openliberty.microprofile.telemetry.1.0.internal_fat,io.openliberty.microprofile.telemetry.1.0.internal_fat_tck,io.openliberty.microprofile.telemetry.1.0.internal.jaeger_fat,io.openliberty.microprofile.telemetry.1.1.internal_fat_tck

File: dev/io.openliberty.microprofile.telemetry.1.0.internal/bnd.bnd [hunk 22]
```diff
@@ -51,17 +50,17 @@ Service-Component: \
 # Need an explicit import for api.common because it gets used by weld proxies
 # created for other API classes
 Import-Package: \
-	io.opentelemetry.api.common,\
-	io.openliberty.microprofile.telemetry.common.*,\
-	*
+  com.ibm.websphere.ras,\
+  com.ibm.websphere.ras.annotation,\
+  io.opentelemetry.api.common,\
+  io.openliberty.microprofile.telemetry.internal.common.resources,\
```
Comment (benjamin-confino) [inline dev/io.openliberty.microprofile.telemetry.1.0.internal/bnd.bnd:56]: For future reference, this line is the one that solves #26158

**Comment by Yas-Franco** [comment]
There are full mode tests that expect to fail in this due to [this change](https://github.com/OpenLiberty/open-liberty/commit/65559cf8f0018c2342128dcd9ca25142df3c5a7b#diff-dfe67d716f21ed331ac903c578391e8c947e0617c9d0e9a4a814e667b161f035). They have been failing since but not running as changes to the package do not trigger a full fat run in the pb

**Comment by Yas-Franco** [comment]
As expected, the failures are from the full fat tests that were failing previously

**Comment by benjamin-confino** [comment]
The build has completed, no additional tests failed apart from the ones Yasmin mentioned.

**Comment by benjamin-confino** [comment]
#libby

**Comment by Yas-Franco** [comment]
#26156 passed in FULL mode with these changes

**Review by tevans78 — APPROVED** [review]

**Comment by Azquelt** [comment]
Heads up, you mustn't write `fixes #xxxx and #xxxx` in your PR description - github will only pick up the first link.

It has to be
```
fixes #xxxx
fixes #xxxx
```

I've fixed the description on this PR.
